src/vs/base/browser/ui/button/button.ts

```diff
@@ -137,8 +137,8 @@ export class Button extends Disposable implements IButton {
 
 		// Also set hover background when button is focused for feedback
 		this.focusTracker = this._register(trackFocus(this._element));
-		this._register(this.focusTracker.onDidFocus(() => this.setHoverBackground()));
-		this._register(this.focusTracker.onDidBlur(() => this.applyStyles())); // restore standard styles
+		this._register(this.focusTracker.onDidFocus(() => { if (this.enabled) { this.setHoverBackground(); } }));
+		this._register(this.focusTracker.onDidBlur(() => { if (this.enabled) { this.applyStyles(); } }));
```

Comment:
Hm... Maybe you don't own this... but why is it again that we do this in JS? Sounds like this should all be achieved in CSS, right?

Reply:
I think this was just to avoid generating dynamic CSS in base widgets, but agree this could be done differently. I am not sure if there are base widgets that create style sheets, maybe only tree/list do?

Reply:
Oh I see, it was a themeing decision. But still: the base widget should use simple CSS rules with predefined colors. And the workbench should override those in dynamic CSS, based on the theme changes, right?

Reply:
But widgets in base still have dynamic colors that you can pass in via ctor options, so the widgets need to make this dynamic.

Reply:
@joaomoreno, @bpasero, is there an action item for me in this PR?